File: tests/fixtures/check_then_act/ts/second_read_fixture.ts
// A synchronous call to another system between the check and the write; flagged only when configured.
export function charge(repo: any, billing: any, id: string) {
  const invoice = repo.get(id);
  if (invoice.paid) return; // expect: check
  billing.callBilling(id); // expect: gap second-read
  repo.markPaid(id); // expect: write
}

File: tests/fixtures/check_then_act/py/second_read_fixture.py
# A synchronous call to another system between the check and the write; flagged only when configured.
def charge(repo, billing, invoice_id):
    invoice = repo.get(invoice_id)
    if invoice.paid:  # expect: check
        return
    billing.call_billing(invoice_id)  # expect: gap second-read
    repo.mark_paid(invoice_id)  # expect: write
